process_report/process_report.py

```diff
@@ -195,16 +242,26 @@ def remove_non_billables(dataframe, pi, projects):
     return filtered_dataframe
 
 
-def remove_billables(dataframe, pi, projects, output_file):
+def remove_billables(dataframe, pi, projects, output_file, invoice_month, upload_to_s3):
```

Comment:
Instead of uploading in each of these functions, do all of the uploads in one separate `upload` function at the end, which takes the path and invoice list and uploads it all there.

This way you limit the side effects to the upload part and don't need to change the tests.